core/app/models/spree/payment.rb

```diff
@@ -165,9 +165,16 @@ def source_actions
 
     def validate_source
       if source && !source.valid?
-        source.errors.each do |field, error|
-          field_name = I18n.t("activerecord.attributes.#{source.class.to_s.underscore}.#{field}")
-          errors.add(I18n.t(source.class.to_s.demodulize.underscore, scope: 'spree'), "#{field_name} #{error}")
+        if Rails.version.to_f >= 6.1
```

Comment:
I think the most idiomatic way of doing this is with:

```ruby
Gem::Requirement.new(">= 6.1").satisfied_by?(Rails.gem_version)
```

Reply:
Thank you @kennyadsl, this is a much better way of checking the Rails version requirement. I have implemented your suggestion in the latest commit.